crates/miden-objects/src/transaction/tx_args.rs

```diff
@@ -165,11 +166,10 @@ impl TransactionArgs {
         let script = note_recipient.script();
         let script_encoded: Vec<Felt> = script.into();
 
-        let new_elements = [
-            (note_recipient.digest(), note_recipient.format_for_advice()),
-            (inputs.commitment(), inputs.format_for_advice()),
-            (script.root(), script_encoded),
-        ];
+        let mut new_elements = note_recipient.to_advice_map_entries();
```

Comment:
nit: I would maybe move the logic from `to_advice_map_entries()` into this function. It looks simple enough, and I think this is the only place where we use it.